src/hwProvider.ts

```diff
@@ -13,12 +13,23 @@ import { IHWWalletApp } from "./interface";
 import { compareVersions } from "./versioning";
 
 export class HWProvider {
-    hwApp?: IHWWalletApp;
-    addressIndex: number = 0;
+    constructor(public _addressIndex: number = 0, public _hwApp?: IHWWalletApp) {}
 
-    constructor() {
+    public get addressIndex(): number {
+        return this._addressIndex;
     }
 
+    public set addressIndex(addressIndex: number) {
```

Comment:
Is this for mocking only? Can we remove the setter - since it's behavior would be unexpected for an eventual caller, anyway?

Now we have the async `setAddressIndex`, which actually talks to the device to switch to a new address.